projects/core/src/error-handling/utils/error-interceptor-utils.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ChainedErrorInterceptorFn,
+  ErrorInterceptor,
+  ErrorInterceptorPriority,
+} from '../error-interceptors/error-interceptor';
+
+/**
+ * Sorts error interceptors based on priority.
+ *
+ * @param errorInterceptors - error interceptors to sort
+ * @returns sorted error interceptors
+ *
+ * @internal
+ */
+export const sortErrorInterceptors = (
+  errorInterceptors: ErrorInterceptor[]
+): ErrorInterceptor[] => {
+  const highPriorityErrorInterceptors = new Array<ErrorInterceptor>();
+  const lowPriorityErrorInterceptors = new Array<ErrorInterceptor>();
+  const noPriorityErrorInterceptors = new Array<ErrorInterceptor>();
```

Comment:
[MINOR]
Now this mechanism is fixed to only 3 possible priority values, for no good reason. At the same time, I don't expect anyone using plenty of various interceptor priority values. That said, I'd like our mechanism to be effortlessly open for customers using many custom various priority numbers. Generic solution could use a hashmap like in the example below:
```ts
const interceptorsGroupedByPriority = new Map<ErrorInterceptorPiority,ErrorInterceptors[]>()

function addToProperGroup(interceptor) {
  // 1. create group if empty
  // 2. push to the group
}

// 1. put interceptors to proper groups (preserving the original order within each group):
errorInterceptors.forEach((interceptor => addToProperGroup(interceptor))

// 2. get the list of all different various unique priority values (in case when customer used also values like 3 or -3)

// 3. sort the list of different priorities (e.g. so it becomes `[-2, -1, 0, 1, 2]`
// why? Because Object.keys may potentially return the Map's keys in a random order.

// 4. concat the whole groups next to each other, in the order of their priorites
return
```

Expected example, where H=High, L=LOW, D=Default; numbers are just additional identifiers):
```
input:  [H1, D1, L1, H2, D2, L2]
output: [H1, H2, D1, D2, L1, L2]
```
Please note in the result, the group of `H`s is before the group of `D`s, which is before the group of `L`s. And that the original order within each group is preserved  - `1` is before `2` inside each group

Reply:
We didn't consider such a scenario during discussions so I didn't implement this. We can consider implementing this while keeping in mind it will increase the time spent on this task. I'm ok with that as long as it's ok with other priorities.

Reply:
Actually, such a way of sorting may resolve the problem described in this comment: https://github.com/SAP/spartacus/pull/17865#discussion_r1333989010

Reply:
The sorting has been changed to address comments from [this](https://github.com/SAP/spartacus/pull/17865#discussion_r1333196192) thread